clients/java/src/main/java/io/camunda/zeebe/client/api/search/FlowNodeInstanceState.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.api.search;
+
+import io.camunda.client.protocol.rest.FlowNodeInstanceFilterRequest;
+import io.camunda.client.protocol.rest.FlowNodeInstanceItem;
+
+public enum FlowNodeInstanceState {
+  ACTIVE,
+  COMPLETED,
+  TERMINATED;
+
+  public static FlowNodeInstanceFilterRequest.StateEnum toProtocolState(
+      FlowNodeInstanceState value) {
+    return (value == null) ? null : FlowNodeInstanceFilterRequest.StateEnum.fromValue(value.name());
```

Comment:
🔧 I suggest we handle this consistently in the client regarding error messages in such cases. Looking at the [PartitionInfo handling](https://github.com/camunda/camunda/blob/main/clients/java/src/main/java/io/camunda/zeebe/client/impl/response/PartitionInfoImpl.java#L36-L47), the error message is more user-focused. Let's do this here as well, ideally by creating a utility method for enum conversion that always yields the same error message for consistency.